### PR TITLE
Document workspaces in README

### PR DESCRIPTION
**Pull request by MikeSpreitzer**
Signed-off-by: Mike Spreitzer <mspreitz@us.ibm.com>

## Summary
This PR updates the README with a summary of the current way workspaces appear to users.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -45,6 +45,54 @@ Yes! This work is still in early development, which means it's _not ready for pr
 
  [![asciicast](https://asciinema.org/a/471709.svg)](https://asciinema.org/a/471709)
 
+## Workspaces in v0.5.0
+
+The workspaces are arranged in a tree.  Each vertex in the tree has a name.  The name of the root vertex is `root`, and no other vetex may have that name.  Other vertices can be addressed by a pathname formed by putting all the path's vertex names together, separated by colons.  For example, the pathname `root:default:foo` identifies the child `foo` of the child `default` of the `root`.
+
+The `kubectl kcp workspace` CLI can be used to work with workspaces.  You can use `ws` as a shorthand for `workspace`.
+
+There are four kinds of workspaces, as follows.
```
Comment (sttts) [inline README.md:54]: ... by default in KCP. Any service built on-top of KCP can use different hierarchies. These are just means as examples.

File: README.md [hunk 2]
```diff
@@ -45,6 +45,54 @@ Yes! This work is still in early development, which means it's _not ready for pr
...
+
+- **The root workspace**.  There can be only one.  It can only have children of type Organization.
+- **Organization**.  Two types of children are allowed, Team and Universal.
+- **Team**.  One type of child is allowed, Universal.
+- **Universal**.  No children are allowed.
+
+Following is a typescript showing some workspace traversal and creation.
```
Comment (sttts) [inline README.md:61]: s/typescript/script/ or shell script. Typescript has different meaning.

File: README.md (hunk 1 above)
Comment (MikeSpreitzer) [reply README.md]: Revised.

File: README.md (hunk 2 above)
Comment (MikeSpreitzer) [reply README.md]: Revised.

**Comment by sftim** [comment]
/ok-to-test
right?

**Comment by MikeSpreitzer** [comment]
@sttts : I revised a while ago, can you please take another look?

File: README.md [hunk 3]
```diff
@@ -45,6 +45,54 @@ Yes! This work is still in early development, which means it's _not ready for pr
 
  [![asciicast](https://asciinema.org/a/471709.svg)](https://asciinema.org/a/471709)
 
+## Workspaces in v0.5.0
+
+The workspaces are arranged in a tree.  Each vertex in the tree has a name.  The name of the root vertex is `root`, and no other vetex may have that name.  Other vertices can be addressed by a pathname formed by putting all the path's vertex names together, separated by colons.  For example, the pathname `root:default:foo` identifies the child `foo` of the child `default` of the `root`.
```
Comment (sttts) [inline README.md:50]: I like math, but others struggle to follow if we use math terms to explain. Everybody knows hierarchical file system tree. Can we use that to decribe how workspaces relate to each other?

File: README.md [hunk 4]
```diff
@@ -45,6 +45,54 @@ Yes! This work is still in early development, which means it's _not ready for pr
 
  [![asciicast](https://asciinema.org/a/471709.svg)](https://asciinema.org/a/471709)
 
+## Workspaces in v0.5.0
```
Comment (sttts) [inline README.md:48]: don't like version references. This text should be independent and updated when something changes.

File: README.md [hunk 5]
```diff
@@ -45,6 +45,54 @@ Yes! This work is still in early development, which means it's _not ready for pr
...
+
+The workspaces are arranged in a tree.  Each vertex in the tree has a name.  The name of the root vertex is `root`, and no other vetex may have that name.  Other vertices can be addressed by a pathname formed by putting all the path's vertex names together, separated by colons.  For example, the pathname `root:default:foo` identifies the child `foo` of the child `default` of the `root`.
+
+The `kubectl kcp workspace` CLI can be used to work with workspaces.  You can use `ws` as a shorthand for `workspace`.
```
Comment (sttts) [inline README.md:52]: even `kubectl ws`

File: README.md [hunk 6]
```diff
@@ -45,6 +45,54 @@ Yes! This work is still in early development, which means it's _not ready for pr
...
+
+There are four kinds of workspace built into kcp, as follows.  Any service built on top of kcp can use a different hierarchy.
+
+- **The root workspace**.  There can be only one.  It can only have children of type Organization.
+- **Organization**.  Two types of children are allowed, Team and Universal.
+- **Team**.  One type of child is allowed, Universal.
+- **Universal**.  No children are allowed.
```
Comment (sttts) [inline README.md:59]: these types are being changed while we write this. `Universal` already got the capability to have sub-workspaces.

File: README.md [hunk 7]
```diff
@@ -45,6 +45,54 @@ Yes! This work is still in early development, which means it's _not ready for pr
...
+
+Following is a shell session showing some workspace traversal and creation.
+
+```console
+$ kubectl kcp ws
+Current workspace is "root:default".
```
Comment (sttts) [inline README.md:65]: users now start in root

File: README.md [hunk 8]
```diff
@@ -45,6 +45,54 @@ Yes! This work is still in early development, which means it's _not ready for pr
...
+
+$ kubectl kcp ws ..
+Current workspace is "root".
+
+$ kubectl kcp ws list
```
Comment (sttts) [inline README.md:70]: this command is removed any day. `kubectl get workspaces` is the replacement.

**Comment by MikeSpreitzer** [comment]
See #1584 instead.

**Pull request closed without merging**